From 33c75c796d101dd89f0c8ce241d5c78dfad322d5 Mon Sep 17 00:00:00 2001
From: zhangmeng <zhangmeng.kevin@spacemit.com>
Date: Thu, 23 Nov 2023 19:17:04 +0800
Subject: [PATCH 0161/1206] fix some warning when building dts

  case 1:
    Warning (ranges_format): /soc/pinctrl@d401e000:ranges: empty "ranges" property but its #address-cells (1) differs from /soc (2)
  fix:
    delete 'ranges' property in pinctrl node

  case 2:
    Warning (interrupts_property): /soc/dsi2@d421a800: Missing interrupt-parent
  fix:
    add 'interrupt-parent' property for dsi2 node

  case 3:
    Warning (graph_port): /soc/dpu_online2@c0340000: graph port node name should be 'port'
  fix:
    modiy name of the node from 'dpu_online2' to 'port'

  case 4:
    Warning (reg_format): /soc/qspi@d420c000/flash@0:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 1)
  fix:
    add adress-cells and size-cells property
    '    #address-cells = <1>;'
    '    #size-cells = <0>;'

  case 5:
    Warning (unit_address_format): /reserved-memory/dpu_reserved@000000000: unit name should not have leading 0s
  fix:
    modify the name to 'dpu_resv: dpu_reserved@27000000'

Change-Id: Ied2e8b3bb546852194c9694647de830628fa11b5
---
 arch/riscv/boot/dts/spacemit/k1-pro_pinctrl.dtsi |  0
 arch/riscv/boot/dts/spacemit/k1-x.dtsi           | 12 ++++++------
 arch/riscv/boot/dts/spacemit/k1-x_fpga.dts       |  4 ++--
 3 files changed, 8 insertions(+), 8 deletions(-)
 mode change 100755 => 100644 arch/riscv/boot/dts/spacemit/k1-pro_pinctrl.dtsi

diff --git a/arch/riscv/boot/dts/spacemit/k1-pro_pinctrl.dtsi b/arch/riscv/boot/dts/spacemit/k1-pro_pinctrl.dtsi
old mode 100755
new mode 100644
diff --git a/arch/riscv/boot/dts/spacemit/k1-x.dtsi b/arch/riscv/boot/dts/spacemit/k1-x.dtsi
index d2dc33006806..0327cc6cdc05 100644
--- a/arch/riscv/boot/dts/spacemit/k1-x.dtsi
+++ b/arch/riscv/boot/dts/spacemit/k1-x.dtsi
@@ -237,7 +237,7 @@ opp1600000000 {
 	clocks {
 		#address-cells = <0x2>;
 		#size-cells = <0x2>;
-		ranges;
+
 		vctcxo_24: clock-vctcxo_24 {
 			#clock-cells = <0>;
 			compatible = "fixed-clock";
@@ -336,7 +336,6 @@ soc {
 		compatible = "simple-bus";
 		#address-cells = <2>;
 		#size-cells = <2>;
-		ranges;
 
 		clint0: clint@e4000000 {
 			compatible = "riscv,clint0";
@@ -405,7 +404,6 @@ pinctrl: pinctrl@d401e000 {
 			#size-cells = <1>;
 			#pinctrl-cells = <2>;
 			#gpio-range-cells = <3>;
-			ranges;
 
 			pinctrl-single,register-width = <32>;
 			pinctrl-single,function-mask = <0xff77>;
@@ -720,7 +718,6 @@ usbdrd3: usb3@0 {
 			clock-names = "usbdrd30";
 			interrupt-parent = <&intc>;
 			interrupts = <149>;
-			ranges;
 			status = "disabled";
 
 			dwc3@c0a00000 {
@@ -899,8 +896,10 @@ pcie2_intc: interrupt-controller@0 {
 			};
 		};
 
-		qspi: qspi@d420c000 {
+		qspi: spi@d420c000 {
 			compatible = "spacemit,k1x-qspi";
+			#address-cells = <1>;
+			#size-cells = <0>;
 			reg = <0x0 0xd420c000 0x0 0x1000>,
 			      <0x0 0xb8000000 0x0 0xc00000>;
 			reg-names = "qspi-base", "qspi-mmap";
@@ -976,7 +975,7 @@ display-subsystem {
 			ports = <&dpu_online2>;
 		};
 
-		dpu_online2: dpu_online2@c0340000 {
+		dpu_online2: port@c0340000 {
 			compatible = "spacemit,dpu-online2";
 			interrupt-parent = <&intc>;
 			interrupts = <90>, <89>;
@@ -1007,6 +1006,7 @@ dsi2: dsi2@d421a800 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0 0xD421A800 0 0x200>;
+			interrupt-parent = <&intc>;
 			interrupts = <95>;
 			ip = "synopsys-dhost";
 			dev-id = <2>;
diff --git a/arch/riscv/boot/dts/spacemit/k1-x_fpga.dts b/arch/riscv/boot/dts/spacemit/k1-x_fpga.dts
index 4a1a62b3cd1b..b2b1fe98a173 100644
--- a/arch/riscv/boot/dts/spacemit/k1-x_fpga.dts
+++ b/arch/riscv/boot/dts/spacemit/k1-x_fpga.dts
@@ -165,7 +165,7 @@ reserved-memory {
 		#size-cells = <2>;
 		ranges;
 
-		linux,cma@70000000 {
+		linux,cma@60000000 {
 			compatible = "shared-dma-pool";
 			/* alloc memory from 0x60000000~0x80000000 */
 			alloc-ranges = <0 0x60000000 0 0x20000000>;
@@ -179,7 +179,7 @@ linux,cma@70000000 {
 		};
 
 		/* reserved 384K for dpu, including mmu table(256K) and cmdlist(128K) */
-		dpu_resv: dpu_reserved@000000000 {
+		dpu_resv: dpu_reserved@27000000 {
 			compatible = "shared-dma-pool";
 			reg = <0x0 0x27000000 0x0 0x00060000>;
 			no-map;
-- 
2.47.0

